File: src/main/java/me/miki/shindo/management/profile/ProfileIcon.java
package me.miki.shindo.management.profile;

import lombok.Getter;
import me.miki.shindo.utils.animation.simple.SimpleAnimation;
import net.minecraft.util.ResourceLocation;

@Getter
public enum ProfileIcon {
    COMMAND(0, "command"), CRAFTING_TABLE(1, "crafting_table"), FURNACE(2, "furnace"), GRASS(3, "grass"),
    HAY(4, "hay"), PUMPKIN(5, "pumpkin"), TNT(6, "tnt");

    private final SimpleAnimation animation = new SimpleAnimation();

    private final int id;
    private final ResourceLocation icon;

    ProfileIcon(int id, String name) {
        this.id = id;
        this.icon = new ResourceLocation("shindo/icons/" + name + ".png");
    }

    public static ProfileIcon getIconById(int id) {

        for (ProfileIcon pi : ProfileIcon.values()) {
            if (pi.getId() == id) {
                return pi;
            }
        }

        return ProfileIcon.GRASS;
    }

}

File: src/main/kotlin/me/miki/shindo/management/profile/ProfileIcon.kt
package me.miki.shindo.management.profile

import me.miki.shindo.utils.animation.simple.SimpleAnimation
import net.minecraft.util.ResourceLocation

enum class ProfileIcon(val id: Int, val icon: ResourceLocation) {
    COMMAND         (0, ResourceLocation("shindo/icons/command.png")),
    CRAFTING_TABLE  (1, ResourceLocation("shindo/icons/crafting_table.png")),
    FURNACE         (2, ResourceLocation("shindo/icons/furnace.png")),
    GRASS           (3, ResourceLocation("shindo/icons/grass.png")),
    HAY             (4, ResourceLocation("shindo/icons/hay.png")),
    PUMPKIN         (5, ResourceLocation("shindo/icons/pumpkin.png")),
    TNT             (6, ResourceLocation("shindo/icons/tnt.png"));

    val animation: SimpleAnimation = SimpleAnimation()

    companion object {
        @JvmStatic
        fun getIconById(id: Int): ProfileIcon = values().firstOrNull { it.id == id } ?: GRASS
    }
}
